lib/src/analytics_logger_generator.dart

```diff
@@ -52,57 +44,56 @@ class AnalyticsLoggerGenerator extends GeneratorForAnnotation<AnalyticsLogger> {
       throw Exception(e);
     }
 
+    // Check if headerRows contains logger separator row
+    for(String loggerSeparatorKey in loggers.keys) {
+      if (!headerRows.contains(loggerSeparatorKey)) {
+        throw Exception('Header row does not contain logger separator row: $loggerSeparatorKey');
+      }
+    }
+
     // enum AnalyticsEvents
     buffer.writeln('enum AnalyticsEvents {');
     for (int i = 0; i < bodyRows.length; i++) {
-      String? _eventSnakeCase =
-          bodyRows[i][headerRows[0]]!.toString().toSnakeCase();
+      String? _eventSnakeCase = bodyRows[i][headerRows[0]]!.toString().toSnakeCase();
       String? camelCaseName = _eventSnakeCase.toCamelCase();
       // TODO(eido9oy): Add support for other analytics providers
-      String outputLine =
-          '$camelCaseName(\'$_eventSnakeCase\'';
-      if (hasFirebaseAnalytics) {
-        String _enableFirebase = bodyRows[i][headerRows[2]] == 'TRUE' ? 'true' : 'false';
-        outputLine += ', $_enableFirebase';
-      }
-      if (hasSingular) {
-        String _enableSingular = bodyRows[i][headerRows[3]] == 'TRUE' ? 'true' : 'false';
-        outputLine += ', $_enableSingular';
+      String outputLine = '$camelCaseName(\'$_eventSnakeCase\'';
+
+      List<String> outputValues = [];
+      // Find logger separator row index in headerRows and get value
+      for(String loggerSeparatorKey in loggers.keys) {
+        print('headerRows : $headerRows');
+        int row = headerRows.indexOf(loggerSeparatorKey);
+        String? _enable = bodyRows[i][headerRows[row]] == 'TRUE' ? 'true' : 'false';
+        outputValues.add(_enable);
       }
-      if (hasDataDog) {
-        String _enableDataDog = bodyRows[i][headerRows[4]] == 'TRUE' ? 'true' : 'false';
-        outputLine += ', $_enableDataDog';
+
+      // Add logger separator row value to outputLine
+      for (int j = 0; j < outputValues.length; j++) {
+        if (outputValues[j] != '') {
+          outputLine += ', ${outputValues[j]}';
+        }
       }
+
       outputLine += ')';
       if (i < bodyRows.length - 1) {
         buffer.writeln('$outputLine,');
       } else {
         buffer.writeln('$outputLine;');
       }
+
+      print('outputLine : $outputLine');
     }
     // TODO(eido9oy): Add support for other analytics providers
```

Comment:
You can delete this kind of todo comments because you resolve this challenges! :)